### ISSUE TITLE
Defaults for boolean flags do not apply to existing environments

### ISSUE THREAD

**Issue by JoeleneG** [launchdarkly/terraform-provider-launchdarkly#114]
Hello!

If you define a boolean flag like it is in the examples but also add the defaults block, it does not obey the defaults values until the second time you run it, by which point the defaults are already set and do not apply to existing environments properly. See:

```
resource "launchdarkly_feature_flag" "boolean_flag" {
  project_key    = launchdarkly_project.tf_flag_examples.key
  key            = "boolean-flag"
  name           = "Bool feature flag"
  description    = "An example boolean feature flag that can be turned either on or off"
  variation_type = "boolean"

  defaults {
    on_variation = 1
    off_variation = 1
  }
}
```

The defaults should be set to false for both, but that does not happen, the on default is set to true in my experience. Running terraform apply a second time acknowledges that it is wrong and switches the defaults to false, however by that point the values in the environments are already set so it does not change anything.

I've done some experimenting and this works how I expect it to:

```
resource "launchdarkly_feature_flag" "boolean_flag" {
  project_key    = launchdarkly_project.tf_flag_examples.key
  key            = "boolean-flag"
  name           = "Bool feature flag"
  description    = "An example boolean feature flag that can be turned either on or off"
  variation_type = "boolean"

  variations {
    value  = true
  } 
  variations {
    value  = false
  } 

  defaults {
    on_variation = 1
    off_variation = 1
  }
}
```

However, it would be nice to not need to include the variations for a boolean flag if possible.

Thanks!

**Comment by ldhenry** [comment]
Hey @JoeleneG,

Thanks for bringing this to our attention. We will look into this issue soon and report back as we learn more.  In the mean time, I'm glad you found a suitable workaround to unblock yourself.

Thanks,
Henry.

**Comment by sloloris** [comment]
Hi @JoeleneG! 

Thank you so much for bringing this to our attention. This was a bug in our provider that has been fixed and released with v2.9.2. Please do let us know if you have any other issues. Have a great day!
